code/builders/builder-vite/src/plugins/csf-plugin.ts

```diff
@@ -10,6 +14,34 @@ export async function csfPlugin(config: Options): Promise<Plugin> {
     // @ts-expect-error - not sure what type to use here
     addons.find((a) => [a, a.name].includes('@storybook/addon-docs'))?.options ?? {};
 
-  // TODO: looks like unplugin can return an array of plugins
-  return vite(docsOptions?.csfPluginOptions) as Plugin;
+  const options = docsOptions.csfPluginOptions;
+
+  return {
+    name: 'vite-plugin-csf',
+    enforce: 'pre',
+    async transform(code, id) {
+      if (!STORIES_REGEX.test(id)) {
```

Comment:
This might be silly but if anyone uses any other pattern they want for story file names, this will bail. Not that they should do that, as most of Storybook's tools only detect stories/story format anyway. Just an FYI!

Reply:
That's correct. I have just copied over this code from the csf-plugin. We should revisit this at some point and read the users story glob instead and transform it to a regex pattern.